gem/backends.py

```diff
@@ -148,6 +163,16 @@ def create_user(self, claims):
         username = claims.get("preferred_username")
         email = claims.get("email", "")  # Email is optional
         # We create the user based on the username and optional email fields.
+
+        # If a user already exists with this username
+        # change that user's username
+        for user in self.UserModel.objects.filter(username=username):
+            if user.profile.auth_service_uuid is None:
```

Comment:
Something we encountered previously is that some users don't have profiles (I'm guessing this is admin or QA users)
Any users like that will cause an error here. you might need to catch it or check that they have a profile beforehand.